File: 19. Remove Nth Node From End of List/19. Remove Nth Node From End of List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    
    let firstPtr = head
    let tempPrev = null
    let temp = head
    for( let i = 0;i<n;i++){
        temp = temp.next
    }
    
    let secondPtr = temp
    while(secondPtr!==null){
        tempPrev = firstPtr
        firstPtr = firstPtr.next
        secondPtr = secondPtr.next
    }
    
    if(tempPrev !==null){
        tempPrev.next = firstPtr.next
    }else{
        head = firstPtr.next
    }
    return head
};

File: 19. Remove Nth Node From End of List/19. Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
       
        dummy = ListNode(0,head)
        left_ptr = dummy
        right_ptr= head
        count = 0

        while(right_ptr!=None):
            if(count<n):
                right_ptr=right_ptr.next
            else:
                right_ptr=right_ptr.next
                left_ptr=left_ptr.next
            
            count+=1

        left_ptr.next=left_ptr.next.next

        return dummy.next
